nni/algorithms/compression/v2/pytorch/base/compressor.py

```diff
@@ -0,0 +1,285 @@
+                    if op_type == 'default':
+                        expanded_op_types.extend(weighted_modules)
+                    else:
+                        expanded_op_types.append(op_type)
+                config['op_types'] = expanded_op_types
+
+            # check if condition is satisified
+            if 'op_types' in config and layer.type not in config['op_types']:
+                continue
+            if 'op_names' in config and layer.name not in config['op_names']:
+                continue
+
+            ret = config
+        if ret is None or 'exclude' in ret:
+            return None
+        return ret
+
+    def get_modules_wrapper(self) -> OrderedDict[str, Module]:
+        """
+        Returns
+        -------
+        OrderedDict[str, Module]
+            An ordered dict, key is the name of the module, value is the wrapper of the module.
+        """
+        return self.modules_wrapper
+
+    def _wrap_model(self):
+        """
+        Wrap all modules that needed to be compressed.
+        """
+        for _, wrapper in reversed(self.get_modules_wrapper().items()):
+            _setattr(self.bound_model, wrapper.name, wrapper)
+        self.is_wrapped = True
+
+    def _unwrap_model(self):
+        """
+        Unwrap all modules that needed to be compressed.
+        """
+        if self.is_wrapped:
+            for _, wrapper in self.get_modules_wrapper().items():
+                _setattr(self.bound_model, wrapper.name, wrapper.module)
+            self.is_wrapped = False
+
+    def set_wrappers_attribute(self, name: str, value: Any):
+        """
+        To register attributes used in wrapped module's forward method.
+        If the type of the value is Torch.tensor, then this value is registered as a buffer in wrapper,
+        which will be saved by model.state_dict. Otherwise, this value is just a regular variable in wrapper.
+
+        Parameters
+        ----------
+        name
+            Name of the variable.
+        value
+            Value of the variable.
+        """
+        for wrapper in self.get_modules_wrapper():
+            if isinstance(value, torch.Tensor):
+                wrapper.register_buffer(name, value.clone())
+            else:
+                setattr(wrapper, name, value)
+
+    def generate_graph(self, dummy_input: Tensor) -> TorchModuleGraph:
+        """
+        Generate a `TorchModuleGraph` instance of `self.bound_model` based on `jit.trace`.
+
+        Parameters
+        ----------
+        dummy_input
+            The dummy input for `jit.trace`, users should put it on right device before pass in.
+
+        Returns
+        -------
+        TorchModuleGraph
+            A `TorchModuleGraph` instance.
+        """
+        self._unwrap_model()
+        graph = TorchModuleGraph(model=self.bound_model, dummy_input=dummy_input)
+        self._wrap_model()
+        return graph
+
+    def generate_module_groups(self) -> Dict[int, List[str]]:
```

Comment:
Is this function only used by pruner, or also might be used by other successor classes like quantizer?

Reply:
I'm not sure, I place this function here because it needs to traverse the entire model.